packages/office-addin-manifest/src/manifestHandler/manifestHandlerJson.ts

```diff
@@ -1,29 +1,69 @@
-  async modifyManifest(guid?: string, displayName?: string): Promise<any> {
-    throw new Error("Manifest cannot be modified in .json files");
+  async modifyManifest(guid?: string, displayName?: string): Promise<TeamsAppManifest> {
+    const teamsAppManifest: TeamsAppManifest = await JSON.parse(this.fileData);
+
+    if (typeof guid !== "undefined") {
+      if (!guid || guid === "random") {
+        guid = uuidv4();
+      }
+      teamsAppManifest.id = guid;
+    }
+
+    if (typeof displayName !== "undefined") {
+      teamsAppManifest.name.short = displayName;
+    }
+    return teamsAppManifest;
   }
 
   async parseManifest(): Promise<ManifestInfo> {
-    const manifestInfo: ManifestInfo = new ManifestInfo();
-
     try {
-      const file = await JSON.parse(this.fileData);
+      const file: TeamsAppManifest = await JSON.parse(this.fileData);
+      return this.translateTeamsAppManifestToManifestInfo(file);
     } catch (err) {
       throw new Error(`Unable to read data for manifest file: ${this.manifestPath}. \n${err}`);
     }
+  }
+
+  translateTeamsAppManifestToManifestInfo(teamsAppManifest: TeamsAppManifest): ManifestInfo {
+    const manifestInfo: ManifestInfo = new ManifestInfo();
+    manifestInfo.id = teamsAppManifest.id;
+    manifestInfo.appDomains = teamsAppManifest.validDomains;
+    manifestInfo.defaultLocale = teamsAppManifest.localizationInfo?.defaultLanguageTag;
+    manifestInfo.description = teamsAppManifest.description.short;
+    manifestInfo.displayName = teamsAppManifest.name.short;
+    manifestInfo.highResolutionIconUrl = teamsAppManifest.icons.color;
+    // manifestInfo.hosts = teamsAppManifest.extension.requirements.scopes; // We need for the new schema to be published
+    manifestInfo.iconUrl = teamsAppManifest.icons.color;
+    // manifestInfo.officeAppType = teamsAppManifest.extension.requirements.capabilities.name; // We need for the new schema to be published
+    // manifestInfo.permissions = teamsAppManifest.Permissions; // Still need to find an example where Permissions work. Or the manifest doesn't have it, or it is a wrong vlaue.
+    manifestInfo.providerName = teamsAppManifest.developer.name;
+    manifestInfo.supportUrl = teamsAppManifest.developer.websiteUrl;
+    manifestInfo.version = teamsAppManifest.version;
 
-    throw new Error("Manifest cannot be parsed in .json files");
+    return manifestInfo;
   }
 
-  async writeManifestData(manifestData: any): Promise<void> {
-    throw new Error("Manifest cannot be written in .json files");
+  async writeManifestData(manifestData: TeamsAppManifest): Promise<void> {
+    await writeToPath(this.manifestPath, manifestData);
+    await this.updateFileData();
   }
   /* eslint-enable @typescript-eslint/no-unused-vars */
 }
+
+/**
+ * Save manifest to .json file
+ * @param filePath path to the manifest.json file
+ */
+async function writeToPath(path: string, manifest: TeamsAppManifest): Promise<void> {
```

Comment:
Isn't this going to be part of the teams-lib?

Reply:
Yeah, this is the exact copy of the team's lib. I'm just copying-pasting, for now, to be able to develop against it.